projects/storefrontlib/src/cms-components/product/product-images/product-images.component.ts

```diff
@@ -1,68 +1,76 @@
 import { ChangeDetectionStrategy, Component } from '@angular/core';
 import { Product } from '@spartacus/core';
-import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
-import { filter, map } from 'rxjs/operators';
+import { CarouselItem } from 'projects/storefrontlib/src/shared';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { distinctUntilChanged, filter, map, tap } from 'rxjs/operators';
 import { CurrentProductService } from '../current-product.service';
 
-const WAITING_CLASS = 'is-waiting';
-
 @Component({
   selector: 'cx-product-images',
   templateUrl: './product-images.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ProductImagesComponent {
-  waiting: HTMLElement;
-
-  product$: Observable<Product> = this.currentProductService
-    .getProduct()
-    .pipe(filter(Boolean));
-
-  private _imageContainer$ = new BehaviorSubject(null);
+  mainMediaContainer = new BehaviorSubject(null);
 
-  imageContainer$ = combineLatest(this.product$, this._imageContainer$).pipe(
-    map(([product, container]) =>
-      container
-        ? container
-        : product.images && product.images.PRIMARY
-        ? product.images.PRIMARY
-        : {}
+  private product$: Observable<
```

Comment:
Also just thought about this, since the `product$` variable is only used for the `thumbs$` lower could it just be `thumbs$` here and you add the map at the end of this pipe?

Reply:
I prefer to keep it, so it's more clear that the tap on the product is creating the initial container.